pyimzml/ImzMLWriter.py

```diff
@@ -123,43 +133,69 @@ def __init__(self, output_filename, mz_dtype=np.float64, intensity_dtype=np.floa
         self.imzml_template = self.wheezy_engine.get_template('imzml')
 
         self.spectra = []
+        self.hashes = {} #set this to None to force processed mode.
```

Comment:
I had not considered the possibility of a hash collision; thanks for pointing that out. Just for fun I tried with about 115,000 m/z arrays and got 2 collisions. So you are right, some check is needed. 

I think that reading back from disk will be far too slow. For true continuous data this check will have to be run for *every pixel*.

I suggest we add the sum and length to the hash to reduce collisions (`hash_mz = "%s-%s-%s"%(hash(mzs), sum(mzs), len(mzs))`). For 1,217,463 mz arrays this has no collisions on my computer and doubles the calculations time.

Or we could use a longer (more processor intensive) hash. sha256 is already imported so seems like an obvious choice. For 115k mz arrays this has no collisions on my computer and increases the calculation time by a factor of 20.

Or we could write code that keeps the x (maybe 10) most recently checked mz arrays in memory and reads from disk the ones that are not in memory. Depending on the data this could be very slow, but for the majority of users (either true continuous or true processed) this would be the fastest option. It would also be the most complicated to implement, but I don't mind that.

What do you think is the best choice?